props/blaster.h

```diff
@@ -77,10 +83,53 @@ class Blaster : public PropBase {
       } else {
         hybrid_font.PlayCommon(&SFX_blast);
       }
+
+      shots_fired_++;
 #endif
     }
   }
 
+  virtual void Reload() {
+    shots_fired_ = 0;
+
+#ifdef ENABLE_AUDIO
+    if (SFX_reload.files_found()) {
+      hybrid_font.PlayCommon(&SFX_reload);
+    }
+#endif
+  }
+
+  virtual void ClipOut() {
+    if (max_shots_ != -1) shots_fired_ = max_shots_;
+
+#ifdef ENABLE_AUDIO
+    if (SFX_clipout.files_found()) {
+      hybrid_font.PlayCommon(&SFX_clipout);
+    }
+#endif
+  }
+
+  virtual void ClipIn() {
+#ifdef ENABLE_AUDIO
+    if (SFX_clipin.files_found()) {
+      hybrid_font.PlayCommon(&SFX_clipin);
+    }
+#endif
+  }
+
+  // Pull in parent's SetPreset, but turn the blaster on.
+  virtual void SetPreset(int preset_num, bool announce) override {
+    PropBase::SetPreset(preset_num, announce);
+
+    if (!SaberBase::IsOn()) {
```

Comment:
I think we should only do this if there is no "poweron" sound in the font, what do you think?

Reply:
It's easy to add an SFX_poweron check here though.

Reply:
Ah so you mean wrap everything after the parent's SetPreset inside an if (!SFX_poweron) Check. Got it.

Reply:
yes, please do.

Reply:
Does the SFX_poweron Check still even need to happen? With the fix inside PlayMonophonic I am just calling On() now.

Reply:
It would make it possible to provide an poweron.wav / poweroff.wav and wire up a power button, which seems like it might be nice?

Reply:
Ah. Yeah that makes sense.

I should add the explicit BUTTON_POWER to trigger On() too then.

Reply:
I think so.

Reply:
Done.